beacon-chain/state/stateutil/slice_root.go

```diff
@@ -0,0 +1,41 @@
+package stateutil
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v5/encoding/ssz"
+)
+
+// SliceRoot computes the root of a slice of hashable objects.
+func SliceRoot[T ssz.Hashable](slice []T, limit uint64) ([32]byte, error) {
```

Comment:
This is the same logic exactly from beacon-chain/state/stateutil/historical_summaries_root.go, but parameterized / generic version